codegen/module.go

```diff
@@ -1022,6 +1022,134 @@ func (system *ModuleSystem) getSpec(instance *ModuleInstance) (interface{}, bool
 	return spec, true, nil
 }
 
+func (system *ModuleSystem) tryResolveIncrementalBuild(
+	instances []ModuleDependency,
+	resolvedModules map[string][]*ModuleInstance,
+) (map[string][]*ModuleInstance, error) {
+
+	toBeBuiltModules := make(map[string]map[string]*ModuleInstance, 0)
+	for _, className := range system.classOrder {
+		toBeBuiltModules[className] = make(map[string]*ModuleInstance, 0)
+	}
+
+	for _, classOrder := range system.classOrder {
+		classInstances := resolvedModules[classOrder]
+		for _, classInstance := range classInstances {
+			spec, ok, err := system.getSpec(classInstance)
+			if err != nil {
+				return nil, err
+			}
+			if ok {
+				classInstance.genSpec = spec
+			}
+
+			found := false
+			for _, instance := range instances {
+				if instance.ClassName == classInstance.ClassName && instance.InstanceName == classInstance.InstanceName {
+					found = true
+					break
+				}
+			}
+			if !found {
+				fmt.Printf(
+					"Skipping generation of %q %q class of type %q "+
+						"as not needed for incremental build\n",
+					classInstance.InstanceName,
+					classInstance.ClassName,
+					classInstance.ClassType,
+				)
+				continue
+			}
+
+			toBeBuiltModules[classInstance.ClassName][classInstance.InstanceName] = classInstance
+		}
+
+		// Collect things of the same class that depend on us
```

Comment:
This comment is inaccurate. You are collecting *all* things that depend (directly or indirectly) on something in `toBeBuiltModules`.